manim/mobject/types/vectorized_mobject.py

```diff
@@ -906,6 +906,185 @@ def __init__(self, *vmobjects, **kwargs):
+            
+            mob = value
+            if self.show_keys:
+                # This import is here and not at the top to avoid circular import
+                from ...mobject.svg.tex_mobject import TextMobject
+                key_text = TextMobject(str(key)).next_to(value, LEFT)
+                mob.add(key_text)
+            
+            self.submob_dict[key] = mob
+            super().add(value)
+        return self
+
+    def remove(self, key):
+        '''Removes the mobject from the :class:`VDict` object having the key `key`
+        
+            Also, it internally removes the mobject from the `submobjects` :class:`list`
+            of :class:`~.Mobject`, (which is responsible for removing it from the screen)
+
+            Parameters
+            ----------
+            key : 
+                The key of the submoject to be removed
+
+            Returns
+            -------
+            self : :class:`VDict`
+                Returns the :class:`VDict` object on which this method was called
+
+            Examples
+            --------
+            Normal usage::
+            my_dict.remove('square')
+
+        '''
+        if key not in self.submob_dict:
+            raise Exception("The given key '%s' is not present in the VDict" %str(key))
+        super().remove(self.submob_dict[key])
+        del self.submob_dict[key]
+        return self
+
+    def __getitem__(self, key):
+        '''Overriding the [] operator for getting submobject by key
+        
+            Parameters
+            ----------
+            key : any suitable type of a python `dict`
+                The key of the submoject to be accessed
+            
+            Returns
+            -------
+            submob : :class:`VMobject`
+                The submobject corresponding to the key `key`
+
+            Examples
+            --------
+            Normal usage::
+                self.play(ShowCreation(my_dict['s']))
+
+        '''
+        submob = self.submob_dict[key]
+        return submob 
+
+    def __setitem__(self, key, value):
+        '''Overriding the [] operator for assigning submobject like a python dict
+
+            Parameters
+            ----------
+            key : any suitable type of a python `dict`
+                The key of the submoject to be assigned
+            value : :class:`VMobject`
+                The submobject to bind the key to
+            
+            Returns
+            -------
+            None
+
+            Examples
+            --------
+            Normal usage::
+                square_obj = Square()
+                my_dict['sq'] = square_obj
+        
+        '''
+        if key in self.submob_dict:
+            self.remove(key)
+        self.add((key, value))
+
+    def get_all_submobjects(self):
+        ''' To get all the submobjects associated with a particular :class:`VDict` object
+        
+            Returns
+            -------
+            submobjects : :class:`dict_values`
```

Comment:
Return values arent named
```suggestion
            :class:`dict_values`
```